implementation/src/main/java/io/smallrye/mutiny/helpers/queues/Queues.java

```diff
@@ -95,17 +106,54 @@ public static <T> Supplier<Queue<T>> unbounded(int size) {
      * @return the queue
      */
     public static <T> Queue<T> createMpscQueue() {
-        return new MpscLinkedQueue<>();
+        if (Infrastructure.useUnsafeForQueues()) {
+            return new MpscLinkedUnpaddedQueue<>();
+        } else {
+            return new MpscLinkedAtomicQueue<>();
+        }
     }
 
     /**
-     * Create a queue of a strict fixed size.
+     * Creates an unbounded single producer / single consumer queue.
      *
-     * @param size the queue size
+     * @param size the chunk size
+     * @return the queue
+     * @param <T> the item type
+     */
+    public static <T> Queue<T> createSpscUnboundedQueue(int size) {
+        if (Infrastructure.useUnsafeForQueues()) {
+            return new SpscUnboundedUnpaddedArrayQueue<>(size);
+        } else {
+            return new SpscUnboundedAtomicArrayQueue<>(size);
+        }
+    }
+
+    /**
+     * Create a MPSC queue with a given size
+     *
+     * @param size the queue size, will be rounded
      * @param <T> the elements type
      * @return a new queue
      */
-    public static <T> Queue<T> createStrictSizeQueue(int size) {
-        return new ArrayBlockingQueue<>(size);
+    public static <T> Queue<T> createMpscArrayQueue(int size) {
+        if (Infrastructure.useUnsafeForQueues()) {
+            return new MpscUnpaddedArrayQueue<>(size);
+        } else {
+            return new MpscAtomicArrayQueue<>(size);
+        }
+    }
+
+    /**
+     * Check when a non-strictly sized queue overflow.
+     *
+     * @param queue the queue
+     * @param limit the limit, a negative value assumes an unbounded queue
+     * @return {@code true} if the queue overflow, {@code false} otherwise
+     */
+    public static boolean isOverflowing(Queue<?> queue, int limit) {
+        if (limit < 0) {
+            return false;
+        }
+        return queue.size() >= limit;
```

Comment:
queue.size with the linked variant cost O(n) while with the array based (including chunked and unbounded) is O(1)ish (because it keep on doing till the values are stable, but usually converge quite fast, unless on containers).

Reply:
Moving to array variants only 👍